### PR TITLE
WIP: started on betting mechanic as part of the game mechanic

### PR DESCRIPTION
**Pull request by pllearns**

### CODE REVIEW

File: src/currency.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = function formatAsMoney(dollars) {
+  if ( typeof dollars !== 'number' ) throw new Error( 'expected ('+JSON.stringify(dollars)+') to be a number')
+  return '$'+dollars.toFixed(2)
+}
```
Comment (jrobcodes) [inline src/currency.js]: I would clean this up a little with ES6, and write clearer code with whitespace and blocks:

``` es6
const formatAsMoney = dollars => {
  if( typeof dollars !== 'number' ) {
    throw new Error( `Expected ${JSON.stringify( dollars )} to be a number` )
  }

  return `$${dollars.toFixed( 2 )}`
}

export default formatAsMoney
```

File: src/game.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+const _ = require('lodash')
+const Betting = require('./betting')
+const currency = require('./currency')
+
+module.exports = class Game {
+  constructor() {
+    this.minBet = 10;
+    this.maxBet = 500;
+    this.CBet = 10;
```
Comment (jrobcodes) [inline src/game.js]: I would consider more intention revealing variable names - `CBet` does not convey any meaningful clue as to its purpose (nor do `minCBet`, `midCBet`, etc. )

File: src/game.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
+const _ = require('lodash')
```
Comment (jrobcodes) [inline src/game.js]: If we're using ES6, we should consistently use the ES6 import syntax:

``` es6
import _ from 'lodash'
import Betting from './betting'
import currency from './currency'
```

File: src/game.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+const Betting = require('./betting')
+const currency = require('./currency')
+
+module.exports = class Game {
+  constructor() {
+    this.minBet = 10;
```
Comment (jrobcodes) [inline src/game.js]: The terminating semicolon is inconsistently used in your code; I prefer omitting them entirely, but please be consistent.

File: src/game.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    return prompt.forString(ask)
+  }
+
+  promptForNumber(ask){
+    return prompt.forNumber(ask)
+  }
+
+  setup() {
+    clear()
+    if (!this.numberOfHumanPlayers){
```
Comment (jrobcodes) [inline src/game.js]: I prefer to use explicit boolean expressions - `if( ! this.numberOfHumanPlayers == 2 )` (or whatever it is supposed to be).

File: src/game.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      this.numberOfHumanPlayers = this.promptForNumber("How may Risk Takers?")
+      this.report('okay', this.numberOfHumanPlayers+' victims')
+      this.createHumanPlayers();
```
Comment (jrobcodes) [inline src/game.js]: Another inconsistent use of semi-colon...

File: src/game.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    if (!this.numberOfAiPlayers){
```
Comment (jrobcodes) [inline src/game.js]: Same as above, I would consider the use of more explicit boolean expressions.

File: src/game.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+  prompt(ask){
+    return prompt.forString(ask)
+  }
+
+  promptForNumber(ask){
+    return prompt.forNumber(ask)
+  }
+
+  setup() {
```
Comment (jrobcodes) [inline src/game.js]: There's a lot going on in the `setup` method.  I would consider breaking it up into logical pieces that are doing one small piece of the overall setup to increase clarity (and, I suspect, to reveal opportunities to refactor).

File: src/game.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+  createHumanPlayers() {
+    for (let i=0; i < this.numberOfHumanPlayers; i++){
+      let name = this.prompt("Give victim #"+(i+1)+"a name:")
+      this.players.push(
+        new HumanPlayer ({
+          game: this,
+          name: name
+        })
+      )
+      this.shufflePlayers()
```
Comment (jrobcodes) [inline src/game.js]: This should be moves outside of the for loop - you only need to shuffle players once, after they have all been defined.

File: src/game.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+      game: this,
+    })
+    this.players.push(this.dealer)
+    this.dealers.deck = new Deck(this.players.length)
+    this.hands = []
+  }
+
+  createHumanPlayers() {
+    for (let i=0; i < this.numberOfHumanPlayers; i++){
+      let name = this.prompt("Give victim #"+(i+1)+"a name:")
```
Comment (jrobcodes) [inline src/game.js]:
``` es6
const name = this.prompt( `Give victim #${i + 1} a name:` )
```

File: src/game.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      this.shufflePlayers()
+    }
+  }
+
+  createAiPlayers() {
+    for (let i=0; i < this.numberOfAiPlayers; i++){
+      this.players.push(
+        new AiPlayer ({
+          game: this,
+          name: 'Ai victim'+(i+1),
```
Comment (jrobcodes) [inline src/game.js]:
``` es6
name: `Ai victim ${i + 1}`
```

File: src/game.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+        })
+        this.shufflePlayers()
+      )
+    }
+  }
+
+  shufflePlayers()
+  this.players = _.shuffle(this.players)
```
Comment (jrobcodes) [inline src/game.js]: Indentation is off.

File: src/game.js [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+start(){
+  let playAgain
+  while(true){
```
Comment (jrobcodes) [inline src/game.js]: Add whitespace to increase readability - I typically separate set up from logic.

File: src/game.js [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+    this.round = new Round({
+      game: this,
+      index: this.roundIndex++,
+    })
+
+    prompt.forOption('Would you like to wager your shirt?', {
+      '[y]es': ()=>{ this.minCBet = true},
+      '[n]o': ()=>{ this.minCBet = false},
+    })
+    if (playAgain){ continue }else{ break }
```
Comment (jrobcodes) [inline src/game.js]: Add whitespace, and increase readability by placing blocks on their own line.  As well, this stands out as a code smell to me, and could be replaced with the following:

``` es6
let playAgain = true

while( playAgain ) {
  // All the things
  playAgain = true; // I don't see where this is getting updated right now, but it should be updated here
}
```

**Review by jrobcodes — COMMENTED** [review]
Not sure this addressed your original question about how to bet clothing... I think the ability to bet a thing should be determined by the `HumanPlayer` class, each of which would know how much money it had to bet, as well as how many articles of clothing it had (perhaps in an array).  Then each player could continue to play as long as money > 0 and articlesOfClothing.length > 0.

File: src/game.js (hunk 5 above)
Comment (pllearns) [reply src/game.js]: I am changing these conditions, as the team informed me that they only want one player and one AI Dealer. You will see my updates in the next push

**Pull request closed without merging**